File: eecs268/2020summer/video-lectures/2020.06.08.StackBasics/Stack.cpp

template <typename T>
Stack<T>::Stack()
{
	m_top = nullptr;
}

template <typename T>
Stack<T>::push(T entry)
{
	//empty stack
	if(m_top == nullptr)
	{
		m_top = new Node<T>(entry);
	}
	//non-empty stack
	else
	{
		//Get another pointer to point
		//at the top node
		Node<T>* temp = m_top;
		
		//Use top to point a new node with
		//the new entry
		m_top = new Node<T>(entry);
		
		//connect the new top to the old top
		m_top->setNext(temp);
	}
}

File: eecs268/2022spring/MWF/2022.02.11.stacks_and_queues/stack.py
class Stack:
    #assume __init__ and push

    def pop(self):
        if self._top != None:
            self._top  = self._top.next
        else:
            raise RuntimeError('Stack empty')

    def peek(self):
        if self._top != None:
            return self._top.entry
        else:
            raise RuntimeError('Stack empty')
